crates/db/types/src/traits.rs

```diff
@@ -359,11 +350,67 @@ pub trait L1BroadcastDatabase: Send + Sync + 'static {
     fn get_last_tx_entry(&self) -> DbResult<Option<L1TxEntry>>;
 }
 
+// =============================================================================
+// MMR Database Traits
+// =============================================================================
+
+/// Unified MMR database trait using MmrId for identification
+///
+/// This trait provides a simpler, more direct API where the MMR instance
+/// is identified by an `MmrId` enum rather than a generic scope parameter.
+/// This is the preferred interface for new code.
+pub trait UnifiedMmrDatabase: Send + Sync + 'static {
+    /// Append a new leaf to the specified MMR
+    fn append_leaf(&self, mmr_id: MmrId, hash: [u8; 32]) -> DbResult<u64>;
+
+    /// Get a node hash by MMR position
+    fn get_node(&self, mmr_id: MmrId, pos: u64) -> DbResult<[u8; 32]>;
+
+    /// Get the total MMR size
+    fn mmr_size(&self, mmr_id: MmrId) -> DbResult<u64>;
+
+    /// Get the total number of leaves
+    fn num_leaves(&self, mmr_id: MmrId) -> DbResult<u64>;
+
+    /// Get the individual peak roots
+    fn peak_roots(&self, mmr_id: MmrId) -> Vec<[u8; 32]>;
```

Comment:
These are not verby names which implies that they're all accessors.

Shouldn't `peak_roots` return a result?